src/main/java/org/spongepowered/mod/command/MinecraftCommandWrapper.java

```diff
@@ -0,0 +1,159 @@
+        if (testPermission(source)) {
+            net.minecraftforge.event.CommandEvent event = new net.minecraftforge.event.CommandEvent(this.command, mcSender, args);
+            if (net.minecraftforge.common.MinecraftForge.EVENT_BUS.post(event)) {
+                if (event.exception != null) {
+                    com.google.common.base.Throwables.propagateIfPossible(event.exception);
+                }
+                return false;
+            }
+
+            if (i > -1) {
+                @SuppressWarnings("unchecked")
+                List<Entity> list = PlayerSelector.matchEntities(mcSender, args[i], Entity.class);
+                String s2 = args[i];
+                mcSender.setCommandStat(CommandResultStats.Type.AFFECTED_ENTITIES, list.size());
+
+                for (Entity entity : list) {
+                    args[i] = entity.getUniqueID().toString();
+
+                    if (handler.tryExecute(mcSender, args, this.command, arguments)) {
+                        ++j;
+                    }
+                }
+                args[i] = s2;
+            } else {
+                mcSender.setCommandStat(CommandResultStats.Type.AFFECTED_ENTITIES, 1);
+
+                if (handler.tryExecute(mcSender, args, this.command, arguments)) {
+                    ++j;
+                }
+            }
+        } else {
+            source.sendMessage(Texts
+                    .builder(SpongeMod.instance.getGame().getRegistry().getTranslationById(TRANSLATION_NO_PERMISSION).get(), new Object[0])
+                    .color(TextColors.RED)
+                    .build());
+        }
+
+        mcSender.setCommandStat(CommandResultStats.Type.SUCCESS_COUNT, j);
+        return j > 0;
+    }
+
+    public int getPermissionLevel() {
+        return this.command instanceof CommandBase ? ((CommandBase) this.command).getRequiredPermissionLevel() : -1;
+    }
+
+    public String getCommandPermission() {
+        return this.activeMod.getModId().toLowerCase() + ".command." + this.command.getCommandName();
+    }
+
+    @Override
+    public boolean testPermission(CommandSource source) {
+        if (source instanceof MCSenderWrapper) {
+            return source.hasPermission(getCommandPermission());
+        } else {
+            return this.command.canCommandSenderUseCommand((ICommandSender) source);
+        }
+    }
+
+    @Override
+    public Optional<String> getShortDescription() {
+        return Optional.absent();
+    }
+
+    @Override
+    public Optional<String> getHelp() {
+        return Optional.absent();
+    }
+
+    @Override
+    public String getUsage() {
+        return this.command.getCommandUsage(null);
+    }
+
+    @Override
+    public List<String> getSuggestions(CommandSource source, String arguments) throws CommandException {
+        final List<String> retList = new ArrayList<String>();
+        this.command.addTabCompletionOptions((ICommandSender) source, arguments.split(" "), null);
+        return retList;
```

Comment:
`retList` is not actually used so it will always be empty. Should it be passed into the call to `addTabCompletionOptions`?